lib/wang_matrix/grid.rb

```diff
@@ -5,12 +5,12 @@ def initialize(width: 5, height: 5)
       @grid = (0..(height-1)).map { Array.new(width) }
       @width = width
       @height = height
-      init_with_spaces
+      init_with_empty_tiles
     end
 
     def self.new_from_array(array)
-      height = array.max {|obj| obj.pos.y}.pos.y+1
-      width = array.max {|obj| obj.pos.x}.pos.x+1
+      height = array.max {|obj1, obj2| obj1.pos.y <=> obj2.pos.y}.pos.y+1
+      width = array.max {|obj1, obj2| obj1.pos.x <=> obj2.pos.x}.pos.x+1
```

Comment:
❓ What's this `<=>` all about? Ah...the "spaceship operator" :-) https://stackoverflow.com/questions/827649/what-is-the-ruby-spaceship-operator